debezium-core/src/main/java/io/debezium/relational/history/DatabaseHistoryMetrics.java

```diff
@@ -110,11 +110,13 @@ public void stopped() {
     public void recoveryStarted() {
         status = DatabaseHistoryStatus.RECOVERING;
         recoveryStartTime = Instant.now();
+        LOGGER.info("Started database history recovery");
     }
 
     @Override
     public void recoveryStopped() {
         status = DatabaseHistoryStatus.RUNNING;
+        LOGGER.info("Finished database history recovery of {} changes in {} ms", changesRecovered.get(), Duration.between(Instant.now(), recoveryStartTime).toMillis());
```

Comment:
Stupid question perhaps, but would this be logged as a negative amount, as you do now ... start time?

Reply:
Ok, so indeed the arguments to `between()` must be swapped. I'll fix and merge.

Reply:
Done.